### ISSUE TITLE
Oggf/Flac conf rules do not support downsampling.

### ISSUE THREAD

**Issue by bpa-code** [LMS-Community/slimserver#411]
Current ogg/Flac support does not handle downsampling.

For example, stream http://icecast3.streamserver24.com:18800/motherearth  is a 96kHz  and will not play on a Radio (max 48kHz).

Current convert.conf ogf entry is 
**ogf flc * *
        [flac] --ogg -dcs -- $FILE$ | [flac] -cs --ignore-chunk-sizes --totally-silent --compression-level-0 -** 

The custom-convert.conf to make the stream play is 
**ogf flc * *
        # IFD:{RESAMPLE=-r %d}T:{START=--skip=%t}U:{END=--until=%v}
        [flac]  --ogg -dcs $START$ $END$ -- $FILE$  | [sox] --ignore-length -q -t wav - -t flac -C 0 $RESAMPLE$ -** 

The above has not been tested with files only the stream and also no seeking tests.
Forum thread
https://forums.slimdevices.com/showthread.php?112850-Mother-Earth-Radio-not-playing-on-Squeezebox-Radio

**Comment by mw9** [comment]
> The above has not been tested with files only the stream and also no seeking tests.

LMS' ogg/Flac support is restricted to streams, so perhaps no file test is required.
Original commit here: a8464b42a03621b94a38d578615bd1be5389ac2

Is the ability to seek a requirement ?

**Comment by bpa-code** [comment]
OK. 
I made the comment to cover all the bases as I don't know or use ogf/flac - I was just "solving" a convert.conf problem. Original rule didn't seem to be oriented toward streams.
Are there any ogf/flac podcast / finite length streams ?

**Comment by mw9** [comment]
> Are there any ogf/flac podcast / finite length streams ?

I don't have wide enough experience to be able to say for sure. The only sources I have found which "advertise" ogg/flac streams don't reference any finite length streams, at present. Those sources are:

https://www.hiresaudio.online/cd-quality-internet-radio/
https://community.auralic.com/t/lossless-internet-radio-ogg-flac/852

My uneducated guess is that there are no podcast / finite length streams to be concerned with at present. So, short of setting up a local server for testing purposes, I see little possibility of actually testing the seeking capability of the proposed rule, at least at present.

**Comment by bpa-code** [comment]
I think if someone needs this and suggested fix seems OK, then it should be committed as is.  If somebody has a seek issue - it is clearly flagged here as a potential issue.

**Comment by mw9** [comment]
That sounds right to me.

This issue was also raised in the forum thread _SB Radio no longer plays 24/96 OGF stream_, back in February.
https://forums.slimdevices.com/showthread.php?111597-SB-Radio-no-longer-plays-24-96-OGF-stream

Forum member D1eter posted a rule without seek support which equates to:
```
ogf flc * *
        # IFRD:{RESAMPLE=-r %d}
        [flac] --ogg -dcs -- $FILE$ | [sox] -q --ignore-length -t wav - -t flac -C 0 $RESAMPLE$ -
```
https://forums.slimdevices.com/showthread.php?111597-SB-Radio-no-longer-plays-24-96-OGF-stream&p=963523&viewfull=1#post963523

It apparently worked.

Michael Herger seemed content in principle to take that change up:
> I'd be happy to use this as the standard - if I knew it was working 
> correctly! Would you have a good collection of OGF streams you can test 
> with?

https://forums.slimdevices.com/showthread.php?111597-SB-Radio-no-longer-plays-24-96-OGF-stream&p=963539&viewfull=1#post963539

And there matters stalled ! But D1eter did post a positive result with a collection of four streams:

https://forums.slimdevices.com/showthread.php?111597-SB-Radio-no-longer-plays-24-96-OGF-stream&p=967235&viewfull=1#post967235

**Comment by michaelherger** [comment]
Hi @bpa-code and @mw9 - I don't follow... is this an issue reported? A suggested fix? What do you expect me to do here?

**Comment by bpa-code** [comment]
The problem was highlighted in an August forum post when a stream would not play. I found a solution and was then asked to open an issue.

To record the problem and the fix to convert.conf , I opened this issue.  As the same problem was already noted in Feb - I think some sort of fix needs to be applied.

I am not a user of ogf/flac format so I don't know where it is used (files, streams, podcasts) . The difference between my fix and D1eters is the support of "seek".  I suggest go with minimal fix (D1eter's) and we can update if seek is needed.

**Comment by mw9** [comment]
> I suggest go with minimal fix (D1eter's) and we can update if seek is needed.

I have checked this fix, and it appears to work as required:

The only two 96kHz ogf streams I found play with the fix, but do not play without it:

- JB Radio-2 http://199.189.87.9:10999/flac.m3u
- Mother Earth Radio http://icecast3.streamserver24.com/motherearth.pls
 
A selection of 44.1khz and 48kHz streams play fine with or without the fix.

@bpa-code, do you intend to submit a PR ? I'm happy to oblige if you would like.

**Comment by slartibartfast3** [comment]
> 
> 
> > I suggest go with minimal fix (D1eter's) and we can update if seek is needed.
> 
> I have checked this fix, and it appears to work as required:
> 
> The only two 96kHz ogf streams I found play with the fix, but do not play without it:
> 
>     * JB Radio-2 http://199.189.87.9:10999/flac.m3u
> 
>     * Mother Earth Radio http://icecast3.streamserver24.com/motherearth.pls
> 
> 
> A selection of 44.1khz and 48kHz streams play fine with or without the fix.
> 
> @bpa-code, do you intend to submit a PR ? I'm happy to oblige if you would like.

Was there any further progress with this?

**Comment by philippe44** [comment]
I'm not sure we need the "F" input type, but it probably does not hurt

**Comment by bpa-code** [comment]
@slartibartfast
I haven't retested this fix and there have been changes to transcoding code since Aug. 
I would expect the fix to still work.
Please submit a PR if you want as I won't get around to one in next 2 weeks.

**Comment by slartibartfast3** [comment]
> > I suggest go with minimal fix (D1eter's) and we can update if seek is needed.
> 
> I have checked this fix, and it appears to work as required:
> 
> The only two 96kHz ogf streams I found play with the fix, but do not play without it:
> 
> * JB Radio-2 http://199.189.87.9:10999/flac.m3u
> * Mother Earth Radio http://icecast3.streamserver24.com/motherearth.pls
> 
> A selection of 44.1khz and 48kHz streams play fine with or without the fix.
> 
> @bpa-code, do you intend to submit a PR ? I'm happy to oblige if you would like.

Do you think it is time to submit a PR now? My custom convert.conf still works for Mother Earth Radio.

**Comment by mherger** [comment]
Yes, please.

**Comment by mw9** [comment]
> Do you think it is time to submit a PR now? My custom convert.conf still works for Mother Earth Radio.

Can you confirm what is in your custom-convert.conf ?

**Comment by slartibartfast3** [comment]
> 
> 
> > Do you think it is time to submit a PR now? My custom convert.conf still works for Mother Earth Radio.
> 
> Can you confirm what is in your custom-convert.conf ?

Mine contains

ogf flc * *
	# IFD:{RESAMPLE=-r %d}T:{START=--skip=%t}U:{END=--until=%v}
	[flac]  --ogg -dcs $START$ $END$ -- $FILE$  | [sox] --ignore-length -q -t wav - -t flac -C 0 $RESAMPLE$ -

which I think was the original one that @bpa-code suggested. I see simplified versions have been suggested since then.

**Comment by mw9** [comment]
@slartibartfast3 
I am intending to submit a PR for this simplified version:
```
ogf flc * *
        # IFRD:{RESAMPLE=-r %d}
        [flac] --ogg -dcs -- $FILE$ | [sox] -q --ignore-length -t wav - -t flac -C 0 $RESAMPLE$ -

```
Would you be kind enough to confirm that it does work on your system before I do that.

**Comment by slartibartfast3** [comment]
> 
> 
> @slartibartfast3
> I am intending to submit a PR for this simplified version:
> 
> ```
> ogf flc * *
>         # IFRD:{RESAMPLE=-r %d}
>         [flac] --ogg -dcs -- $FILE$ | [sox] -q --ignore-length -t wav - -t flac -C 0 $RESAMPLE$ -
> ```
> 
> Would you be kind enough to confirm that it does work on your system before I do that.

I can confirm it works for "Mother Earth Radio". I was a bit surprised to see it rebuffering sometimes but that must be unrelated.

**Comment by mw9** [comment]
> I can confirm it works for "Mother Earth Radio". I was a bit surprised to see it rebuffering sometimes but that must be unrelated.

I agree, and was also surprised by the rebuffering. I don't recall that happening last year when the issue was first raised.

PR submitted.

**Comment by mherger** [comment]
Fixed via #544. Thanks!
